pkg/ast/spl/tests/splParser_test.go

```diff
@@ -7793,6 +7795,306 @@ func Test_Format_cmd_Incomplete_RowCol_Options(t *testing.T) {
 	assert.NotNil(t, err)
 }
 
+func getTimeDiff(daylight bool) int64 {
+	localTime := time.Date(2024, time.June, 5, 13, 37, 5, 0, time.Local)
+	if daylight {
+		localTime = time.Date(2024, time.January, 1, 0, 0, 0, 0, time.Local)
+	}
+
+	edtLocation, _ := time.LoadLocation("America/New_York")
+	edtTime := localTime.In(edtLocation)
+
+	_, localOffset := localTime.Zone()
+	_, edtOffset := edtTime.Zone()
+
+	diff := (edtOffset - localOffset)
+	return int64(diff) * 1000
+}
+
+func Test_CalculateRelativeTime_1(t *testing.T) {
+	time := time.Date(2024, time.June, 5, 13, 37, 5, 0, time.Local)
+	// w0
+	rtm := ast.RelativeTimeModifier{
+		Snap: "w0",
+	}
+	epoch, err := spl.CalculateRelativeTime(rtm, time)
+	assert.Nil(t, err)
+	// Sunday, June 2, 2024 12:00:00 AM UTC-4
+	assert.Equal(t, int64(1717300800000)+getTimeDiff(false), epoch)
+}
```

Comment:
nit: you don't need the comment on line 7816 because it's clear from the code right after it, so the comment doesn't add any information. For lines 7822-7823, it would probably be a little better to remove the comment (except for the weekday, since that's important for this particular test) and make a `time.Date` directly using that info and then call `Unix()` to get the unix epoch for that date.

You can keep 7822-7823 how you have it if you want, but you should remove the comment on 7816

Same comment for several of your other tests

Reply:
addressed in the new commit.